client/src/test/java/io/confluent/kafka/schemaregistry/client/rest/RestServiceTest.java

```diff
@@ -52,4 +71,80 @@ public void buildRequestUrl_trimBaseUrlAndPath() {
 
     assertEquals("http://test.com/some/path", RestService.buildRequestUrl(baseUrl, path));
   }
+
+  @Mock
+  private URL url;
+
+  @Mock
+  private HttpURLConnection httpURLConnection;
+
+  @Mock
+  private InputStream inputStream;
+
+  /*
+   * Test setBasicAuthRequestHeader (private method) indirectly through getAllSubjects.
+   */
+  @Test
+  public void testSetBasicAuthRequestHeader() throws Exception {
+    RestService restService = new RestService("http://localhost:8081");
```

Comment:
This is great. But we are asserting more than what's needed in this test method. If we can just verify setAutorizationHeaders and keep the code compact, it would be great. Something like, createNiceMock() could help here.